tiatoolbox/dataloader/wsireader.py

```diff
@@ -836,18 +836,42 @@ def info(self):
             WSIMeta: containing meta information.
 
         """
-        objective_power = np.int(
-            self.openslide_wsi.properties[openslide.PROPERTY_NAME_OBJECTIVE_POWER]
-        )
+        props = self.openslide_wsi.properties
+        if openslide.PROPERTY_NAME_OBJECTIVE_POWER in props:
+            objective_power = int(props[openslide.PROPERTY_NAME_OBJECTIVE_POWER])
+        else:
+            objective_power = None
```

Comment:
Please guess objective from mpp if objective value not available but with a warning to the user.

Reply:
I've added a function to approximate and some tests.